src/remoted/shared_download.c

```diff
@@ -507,6 +507,44 @@ int w_yaml_file_update_structs(){
     return 0;
 }
 
+void w_create_group(char *group){
+       
+    char group_path[PATH_MAX] = {0};
+
+    if(snprintf(group_path,PATH_MAX,isChroot() ? "/etc/shared/%s" : DEFAULTDIR"/etc/shared/%s",group) >= PATH_MAX){
+        minfo(W_PARSER_GROUP_TOO_LARGE,PATH_MAX);
+    }
+    else{
+        /* Check if group exists */
+        DIR *group_dir = opendir(group_path);
+
+        if (!group_dir) {
+            /* Create the group */
+            if(mkdir(group_path,0770) < 0){
+                switch (errno) {
+                    case EEXIST:
+                        if (IsDir(group_path) < 0) {
+                            merror("Couldn't make dir '%s': not a directory.", group_path);
+                        }
+                        break;
+
+                    case EISDIR:
+                        break;
+
+                    default:
+                        merror("Couldn't make dir '%s': %s", group_path, strerror(errno));
+                        break;
+                }
+            } else {
+                if(chmod(group_path,0770) < 0){
+                    merror("Error in chmod setting permissions for path: %s",group_path);
+                }
+            }
+        }
+        closedir(group_dir);
```

Comment:
Please make sure `closedir()` never receives a null pointer in order to prevent segmentation faults:
```
warning: null argument where non-null required (argument 1) [-Wnonnull]
```